pkg/sql/distsql_plan_bulk.go

```diff
@@ -46,9 +47,26 @@ func (dsp *DistSQLPlanner) SetupAllNodesPlanningWithOracle(
 	localityFilter roachpb.Locality,
 ) (*PlanningCtx, []base.SQLInstanceID, error) {
 	if dsp.codec.ForSystemTenant() {
-		return dsp.setupAllNodesPlanningSystem(ctx, evalCtx, execCfg, oracle, localityFilter)
+		return dsp.setupAllNodesPlanningSystem(ctx, evalCtx, execCfg, oracle, localityFilter, nil /*txn*/)
 	}
-	return dsp.setupAllNodesPlanningTenant(ctx, evalCtx, execCfg, oracle, localityFilter)
+	return dsp.setupAllNodesPlanningTenant(ctx, evalCtx, execCfg, oracle, localityFilter, nil /*txn*/)
+}
+
+// SetupAllNodesPlanningWithOracleAndFollowers is like
+// SetupAllNodesPlanningWithOracle but with a txn param that if non-nil allows
+// the flow to plan on followers.
+func (dsp *DistSQLPlanner) SetupAllNodesPlanningWithOracleAndFollowers(
+	ctx context.Context,
+	evalCtx *extendedEvalContext,
+	execCfg *ExecutorConfig,
+	oracle replicaoracle.Oracle,
+	localityFilter roachpb.Locality,
+	txn *kv.Txn,
```

Comment:
Is there a way to pass in a timestamp? Or, if not (because of backport considerations), at least leave a TODO to not rely on txn?

Reply:
Indeed, a timestamp is what I have and would prefer to pass, but we'd need to rip up several layers of API that are shared with all SQL queries to do that, which seems incompatible with the goals of low risk backports.

It is also worth noting that the txn contains a _mutable_ timestamp, that changes over the life of the txn, and the caller in the planning code is reading the _current_ value of it (with appropriate locking), so it probably is not as simple as s/kv.txn/hlc.timestamp/ but rather would need to be something more like s/kv.Txn/func() hlc.timestamp/ and then we'd need to consider if making this func means hot paths start allocating or something. Doable for sure but not the scope I want to introduce in a backportable change.

Reply:
Could we not fix master as correctly as possible, and then think about a backport separately?